Plugin documentation added (#530)
ubccr/coldfront: docs/pages/plugin/how_to_create_a_plugin.md

# How to Create Your Own Plugin For ColdFront

ColdFront plugins are synonymous to Django apps. To create a plugin, all you have to do is create a Django app and link it to ColdFront by editing configuration settings.

> The tutorial below assumes that you are comfortable with Django. If not, check out [these docs]("https://docs.djangoproject.com/en/4.2/intro/tutorial01/") for an introduction to how to build a Django app.

Reviewer: This link doesn't work.  I think you need to take out the quotes around the URL

Throughout this tutorial, to effectively illustrate the structure of a linked ColdFront plugin, the Weekly Report plugin [linked here](https://github.com/rg663/weeklyreportapp) will be used as an example.

## Set Up Your App

1. Create your app, complete with a file structure as demonstrated below:
```
weeklyreportapp
│   README.md  
│   __init__.py
│   admin.py
│   apps.py
│   models.py
│   tests.py
│   urls.py
│   views.py
│
└───templates (you can include as many as needed)
    │   index.html

```

> 2. Ensure that your **__init__.py** file contents have the following format:

Reviewer: ```suggestion
2. Ensure that your app's **__init__.py** file contents have the following format:
```

```
default_app_config = "coldfront.core.weeklyreportapp.apps.WeeklyreportappConfig"
```

> 3. Ensure that your **apps.py** file contents look like this:

Reviewer: ```suggestion
3. Ensure that your app's **apps.py** file contents look like this:
```

```
from django.apps import AppConfig

class WeeklyreportappConfig(AppConfig):
    default_auto_field = 'django.db.models.BigAutoField'
    name = 'coldfront.core.weeklyreportapp'
```

> 4. Make sure to include this line in your **urls.py** file!

Reviewer: ```suggestion
4. Make sure to include this line in your app's **urls.py** file:
```

```
app_name = "weeklyreportapp"
```
## Link Your App to ColdFront

> 1. Add the following app to your list of ```INSTALLED_APPS``` in your ColdFront **base.py** file:

Reviewer: ```suggestion
1. Add the following app to your list of ```INSTALLED_APPS``` in the ColdFront **base.py** (`coldfront/config/base.py`) file:
```

```
# ColdFront Apps
INSTALLED_APPS += [
    'coldfront.core.weeklyreportapp',
]
```

> 2. Edit your main ColdFront **urls.py** file to include the new urls:

Reviewer: ```suggestion
2. Edit the ColdFront **urls.py** (`coldfront/config/urls.py`) file to include the new urls info:
```

    │   index.html

```
2. Ensure that your app's **__init__.py** file contents have the following format:
```
default_app_config = "coldfront.plugins.weeklyreportapp.apps.WeeklyreportappConfig"
```

3. Ensure that your app's **apps.py** file contents look like this:
```
from django.apps import AppConfig

class WeeklyreportappConfig(AppConfig):
    default_auto_field = 'django.db.models.BigAutoField'
    name = 'coldfront.plugins.weeklyreportapp'
```

4. Make sure to include this line in your app's **urls.py** file:
```
app_name = "weeklyreportapp"
```

5. In your app's **views.py** file, add this line:
```
from .models import *
```
   
6. Import ColdFront models in the following manner in your **models.py** file in your app:
```
from coldfront.core.allocation.models import *
from coldfront.core.project.models import *
from coldfront.core.resource.models import *
from coldfront.core.user.models import *
```
   
7. Add ColdFront's skeleton HTML/CSS by adding the following lines to all of your template files:
```
{% extends "common/base.html" %}
{% load crispy_forms_tags %}
{% load humanize %}
{% load static %}
```

## Link Your App to ColdFront

> 1. Add the following app to your list of ```INSTALLED_APPS``` by creating a new file in the ColdFront plugins directory (`coldfront/plugins`) and adding these lines to the new file:

Reviewer: ```suggestion
1. Add the following app to your list of ```INSTALLED_APPS``` by creating a new file in the ColdFront plugins directory (`coldfront/congig/plugins`) and adding these lines to the new file (i.e. `weeklyreportapp.py`):
```
